device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/fileupload/FileUploadTask.java

```diff
@@ -199,9 +208,31 @@ public IotHubStatusCode sendNotification(FileUploadCompletionNotification fileUp
         ResponseMessage responseMessage = httpsTransportManager.sendFileUploadNotification(message);
         httpsTransportManager.close();
 
+        validateServiceStatusCode(responseMessage, "Failed to complete the file upload notification");
```

Comment:
Is this to add additional context to why 403 for example?
Does service not return error message/description?

Reply:
The nested IotHubServiceException is what will contain the important details. At this level, all the method knows is that completing the file upload notification failed